lib/wallet.js

```diff
@@ -124,6 +130,156 @@ class Wallet extends ethers.Wallet {
+        
+        
+        let amountBN = ethers.utils.parseEther(stageAmount);
+        const receiptJSON = receipt.toJSON();
+        if (receiptJSON.currency.ct !== '0x0000000000000000000000000000000000000000') {
+            const tokenInfo = await getTokenInfo.call(this, receiptJSON.currency.ct, true);
+            amountBN = ethers.utils.parseUnits(stageAmount.toString(), tokenInfo.decimals);
+        }
+        
+        const challengeContract = _driipSettlementChallengeContract.get(this);
+        return await challengeContract.startChallengeFromPayment(receipt.toJSON(), this.address, amountBN, options);
+    }
+
+    /**
+     * Settle a payment driip of this wallet.
+     * @param {Receipt} receipt - The receipt object for payment challenge.
+     * @param [options]
+     * @returns {Promise} A promise that resolves into transaction hash.
+     * @example
+     * let hashObj = await wallet.settleDriipAsPayment(receipt, {gasLimit: 200000});
+     */
+    async settleDriipAsPayment(receipt, options = {}) {
+        //if current challenge phrase is in dispute or status is disqualified, not allow to settle
+        const currentPhase = await this.getCurrentPaymentChallengePhase();
+        const currentStatus = await this.getCurrentPaymentChallengeStatus();
+        if (currentPhase === 'Dispute') 
+            throw new Error('Current challenge phase is in dispute, can not settle the driip!');
+        
+        if (currentStatus === 'Disqualified') 
+            throw new Error('Current challenge status is in disqualified, can not settle the driip!');
+        
+        const exchangeContract = _exchangeContract.get(this);
+        return await exchangeContract.settleDriipAsPayment(receipt.toJSON(), this.address, options);
+    }
+
+    /**
+     * Without an amount of ETH from nahmii to base layer.
+     * @param {string} amountEth - The amount of ETH to withdraw from nahmii.
+     * @param [options]
+     * @returns {Promise} A promise that resolves into transaction hash.
+     * @example
+     * let hashObj = await wallet.withdrawEth('1.1', {gasLimit: 200000});
+     */
+    async withdrawEth(amountEth, options = {}) {
+        const currency = '0x0000000000000000000000000000000000000000';
+        const amountWei = ethers.utils.parseEther(amountEth.toString());
+        const clientFund = _clientFund.get(this);
+        return await clientFund.withdraw(amountWei, currency, 0, '', options);
+    }
+
+    /**
+     * Without an amount of ETH from nahmii to base layer.
+     * @param {string} amount - The amount of ETH to withdraw from nahmii.
+     * @param {string} symbol - The token symbol to withdraw.
+     * @param [options]
+     * @returns {Promise} A promise that resolves into transaction hash.
+     * @example
+     * let hashObj = await wallet.withdrawToken('1.1', 'TT1', {gasLimit: 200000});
+     */
+    async withdrawToken(amount, symbol, options = {}) {
```

Comment:
Is it safe to use token symbol? Are there any guarantees that symbol is globally unique?

When making a payment, we use the MonetaryAmount type, which specifies a two component currency descriptor and amount. Would that be better to use here?

Reply:
I actually follow how the `depositToken` handle the symbol, which is looking through the supported tokens and gets the corresponding currency address and decimals for the amount. In this case, we assume the SDK users would check the supported tokens and verify the symbol got the correct address they intend to use.

But I agree with your opinion, that would be safer. I can refactor it to use currency address in the place of symbol variable.

As the `depositToken` was written by @morfj, let's get his confirmation for the change.